app/presenters/schedule_presenter.rb

```diff
@@ -0,0 +1,105 @@
+class SchedulePresenter
+
+  def initialize(venue_object)
+    @venue = venue_object
+  end
+
+  #this is the method actually being called in the view.
+  def shows_at_venue_per_day(week_day)
+    shows_at_venue = Event.where("venue_id" => @venue.id)
+    @shows_at_venue_per_day = []
+    @shows_with_open_time_slots = []
+    shows_at_venue.each do |show|
+      if show.weekday == week_day
+        @shows_at_venue_per_day << show
+      else
+        next
+      end
+    end
+    @shows_at_venue_per_day.sort_by!{ |obj| obj.start_date_time}
+    shows_with_blanks
+    return @shows_with_blanks
+  end
+
+  def shows_with_blanks
+    @shows_with_blanks = []
+    current_time_check = Time.parse("2016-10-06 17:00:00").to_s
+    current_time_check.sub! '-0500', 'UTC'
+    current_time_check = Time.parse(current_time_check)
+    @i = 0
+
+    until @i == @shows_at_venue_per_day.length
+      puts "@shows_at_venue_per_day is #{@shows_at_venue_per_day} It's #{@shows_at_venue_per_day.length} elements in length."
+      start_of_show = @shows_at_venue_per_day[@i].start_date_time
+
+      array_of_events_until_show = find_show_start_time(current_time_check, start_of_show)
+      puts "array_of_events_until_show is #{array_of_events_until_show}"
+      blank_spaces_to_array(array_of_events_until_show)
+      puts "blanks added to array--@shows_with_blanks is #{@shows_with_blanks}"
+      @shows_with_blanks << @shows_at_venue_per_day[@i]
+      puts "show added to array--@shows_with_blanks is #{@shows_with_blanks}"
+
+      @i += 1
+      puts "@i is #{@i}"
+      current_time_check = end_time_of_previous_show
+      puts "current_time_check is #{current_time_check}"
+    end
+  end
+
+  def find_show_start_time(current_time_check, start_of_show)
+    array_of_events_until_show = []
+    empty_block_count = 0
+    until current_time_check == start_of_show
```

Comment:
Refers to current_time_check more than self (maybe move it to another class?) (FeatureEnvy)